### ISSUE TITLE
Improve natural=heath color

### ISSUE THREAD

**Issue by matthijsmelissen** [gravitystorm/openstreetmap-carto#780]
The following issue has been moved over from [trac](https://trac.openstreetmap.org/ticket/3578):

Nyah. I think that current heath color is inharmonious. I propose to replace the color on #ddebbb (see the bottom picture).

**Comment by matkoniecz** [comment]
Difference between natural=heath and natural=scrub is not great. Maybe render both in the same style (using current natural=scrub rendering)? Or in very similar styles like wetland (#1497 by @imagico)?

**Comment by matkoniecz** [comment]
@math1985 

> is the purple=heath rendering a Netherlands-only thing?

heaths in Poland are quite rare, but I remember map or two marking them as purple, some used symbols (symbols were not really rereadable so this map would be a poor source of inspiration).

In Poland typical heath looks like on this image:

![](https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/L%C3%BCneburger_Heide_109.jpg/400px-L%C3%BCneburger_Heide_109.jpg)

maybe it is also true in Netherlands and this is source of using purple colour for marking them?

I thought about using natural=scrub symbol, with plant recoloured to purple. Unfortunately it is not true that typical heath will be associated with purple - see images on https://en.wikipedia.org/wiki/Heath

**Comment by imagico** [comment]
In principle there is of course more or less a continuum between a dense forest and a very sparse heath but in general a typical scrubland and a typical heathland are distinct enough to warrant different rendering.  Abuse of the tags notwithstanding heathlands are distinct habitats characterized by distinct species of scrubs and only exit under certain conditions, they are not just relatively low growing variants of normal scrubland.

Practically of course both natural=scrub and natural=heath are often abused to tag less dense parts of the higher growing vegetation, i.e. natural=scrub is frequently used for open woodlands with scattered but full grown trees and natural=heath is frequently used for grasslands with scattered larger scrubs.  Both are wrong of course.  Ideally there should be tags to document secondary vegetation layers in addition to the dominating type of cover so this could all be addressed properly.

Just for understanding: natural heathlands are mostly limited to maritime and polar/mountain climates, in Europe they occur primarily near the Atlantic coast.  Anthropogenic heaths OTOH also occur in central, eastern and especially southern Europe where human influence limits growth of larger trees and scrubs.

In the eastern Mediterranean for example grazing has lead to sparse heathland being a widely dominating type of vegetation like here:

![heath](http://www.imagico.de/files/heath.jpg)

It would be very wrong and misleading to equate that to the higher growing scrublands typical for the western Mediterranean like here:

![heath](http://www.imagico.de/files/scrub2.jpg)

**Comment by matkoniecz** [comment]
I opened #1733 intended to fix this problem.

**Comment by matkoniecz** [comment]
> In principle there is of course more or less a continuum between a dense forest and a very sparse heath 

Is it also OK to consider heath as something between scrub and grassland? It may lead to something like

![london 17 17 master - heath 300px](https://cloud.githubusercontent.com/assets/899988/9155388/cbfa277c-3eb6-11e5-95be-ddb4481a590c.png)

**Comment by imagico** [comment]
Differences are multi-dimensional here, there are at least:

- the basic vegetation type dimension (woody/herbaceous)
- the height of the vegetation (with the heath/grassland->scrub->wood succession)
- the humid/arid dimension with the extreme case of wetlands - only this is currently specifically mapped in OSM although the different classes imply differences here to some extent.
- the vegetation density (how much of the ground is vegetation covered) - different vegetation tags do not really vary in this dimension.
- the dimension of intensity of human influence.  This is only mapped in case of forest/wood (which does not really work in reality) and grass (grassland indicating little human influence while meadow, possible with meadow=agricultural indicate more and garden/village_green etc. cover the extreme).   For scrub you could consider orchard/vineyard as an intensive human influence variant.  Heath can be fully natural as well as fairly strongly anthropogenic so it also covers a wide range in that dimension.

Since grassland and scrub differ on multiple dimensions if you'd interpolate the color this creates confusion i think.  And in terms of mapper feedback this kind of coloring also encourages abuse as i described (i.e. grassland with occasional scrubs/trees as heath).  And there are only four base colors applying to natural vegetation anyway so this is not a real issue i think.  You could think of creating a color line from hearth via scrub to wood and make grass a completely different tone but i don't think this is feasible considering the other constraints.  So the current system placing scrub and heath on different sides of the line between grass and wood is not bad i think.

**Comment by kocio-pl** [comment]
Do we still want to change the way heath is shown? Violet is used for industrial and power-related areas, so I wouldn't like to use it, but original proposition was like this:

Before
![6ih6oxs](https://user-images.githubusercontent.com/5439713/30245405-c4f35b1c-95d8-11e7-93fa-b4da085b16c4.png)
After
![g5thzond](https://user-images.githubusercontent.com/5439713/30245406-c72e1304-95d8-11e7-8692-df262a75d2d0.png)

**Comment by SomeoneElseOSM** [comment]
FWIW I went for #E6E8C5 for this:
https://github.com/SomeoneElseOSM/openstreetmap-carto-AJT/blob/master/landcover.mss#L49
It's less "in your face" than the current heath colour, but still characteristic.

**Comment by kocio-pl** [comment]
AJT version:
![q8atfc7w](https://user-images.githubusercontent.com/5439713/30247916-4d17661a-961e-11e7-91a6-a68f510d1722.png)

**Comment by kocio-pl** [comment]
AJT color, Ireland - the biggest heath area I know. Z13 looks much better, with water and scrub colors being better visible (as always click to see full images)

z10
Before
![db8nthw](https://user-images.githubusercontent.com/5439713/30249033-2ad9d438-9633-11e7-9e4d-85e7a39952ed.png)
After
![thannq5l](https://user-images.githubusercontent.com/5439713/30249034-2e5bd6ba-9633-11e7-9b30-57a340f4fa6a.png)

z11
Before
![dymitgtw](https://user-images.githubusercontent.com/5439713/30249035-340e85a8-9633-11e7-8bfd-f812ba6929a8.png)
After
![y adzwzv](https://user-images.githubusercontent.com/5439713/30249037-3a60496e-9633-11e7-8c0d-94973ddfbe3c.png)

z12
Before
![uo2nnt98](https://user-images.githubusercontent.com/5439713/30249040-3cbf4c5a-9633-11e7-8d61-b7af0daa1738.png)
After
![eb_vh7v](https://user-images.githubusercontent.com/5439713/30249043-3ff7add6-9633-11e7-8e5b-cc8b7a2137f9.png)

z13
Before
![4ogzjfji](https://user-images.githubusercontent.com/5439713/30249044-437404dc-9633-11e7-92cf-4d49794f296d.png)
After
![4fpqgpg9](https://user-images.githubusercontent.com/5439713/30249045-457de64e-9633-11e7-9460-7bcad3066e0e.png)

**Comment by Tomasz-W** [comment]
@kocio-pl Are you going to do a PR with it? Test renderings looks very good :)

**Comment by kocio-pl** [comment]
I don't plan to, I'm still looking for more coders.

**Comment by kocio-pl** [comment]
Great! It should be checked how would it look like in comparison with a new farmland color, otherwise AJT color was nice.

**Comment by Tomasz-W** [comment]
@Adamant36 Please notice that color proposed above would be quite similar to https://github.com/gravitystorm/openstreetmap-carto/pull/3327 after merging it.

**Comment by Adamant36** [comment]
Before and after with new farmland color and using AJT color. 
before 1
![before 1](https://user-images.githubusercontent.com/30259065/44949767-60577980-aded-11e8-88a2-f7c0ea152468.png)
after 1
![after 1](https://user-images.githubusercontent.com/30259065/44949776-736a4980-aded-11e8-8a35-fa4c487a67a2.png)
before 2
![before 2](https://user-images.githubusercontent.com/30259065/44949778-7b29ee00-aded-11e8-9a33-9980c6dec919.png)
after 2
![after 2](https://user-images.githubusercontent.com/30259065/44949780-82e99280-aded-11e8-9286-16d9b04d743a.png)
before 3
![before 3](https://user-images.githubusercontent.com/30259065/44949781-8aa93700-aded-11e8-93e8-7801fb352912.png)
after 3
![after 3](https://user-images.githubusercontent.com/30259065/44949784-9694f900-aded-11e8-8b3c-6efca0df14e9.png)

**Comment by kocio-pl** [comment]
Thanks for testing, unfortunately they look like shades of the same color for me.

**Comment by Adamant36** [comment]
Yep. I agree. You can barely see the farmland if its in heath.

**Comment by kocio-pl** [comment]
Still there's a room for further searching. Darkness is OK for me, so maybe just something closer to grass for example? There might be a lot of ideas, but testing is the key.

**Comment by Adamant36** [comment]
ddebbb (original color suggested)
![ddebbb](https://user-images.githubusercontent.com/30259065/44984606-43788e80-af32-11e8-8ad7-5531a9350823.png)
c5ebbb
![c5ebbb](https://user-images.githubusercontent.com/30259065/44984613-512e1400-af32-11e8-96e1-f8d71b8796c5.png)
d3e5a7
![d3e5a7](https://user-images.githubusercontent.com/30259065/44984622-5ab77c00-af32-11e8-856f-cf9ca2cb42b2.png)
c9df94
![c9df94](https://user-images.githubusercontent.com/30259065/44984645-673bd480-af32-11e8-8b90-b32688c0ebd0.png)
bfda80
![bfda80](https://user-images.githubusercontent.com/30259065/44984659-74f15a00-af32-11e8-8801-21b96e12931f.png)
bbebbd
![bbebbd](https://user-images.githubusercontent.com/30259065/44984677-80dd1c00-af32-11e8-8564-e9faaeacecca.png)
a7e5aa
![a7e5aa](https://user-images.githubusercontent.com/30259065/44984694-90f4fb80-af32-11e8-8c4a-e9fdba4b6a95.png)
94df97
![94df97](https://user-images.githubusercontent.com/30259065/44984706-9eaa8100-af32-11e8-8d56-b473412360a0.png)
80da84
![80da84](https://user-images.githubusercontent.com/30259065/44984717-abc77000-af32-11e8-8d71-6b857fac3547.png)
Its interesting how the last few almost don't make the woods or the water work out anymore. I like somewhere in the c5ebbb or bbebbd range.

**Comment by dieterdreist** [comment]
Can you explain a bit more about the leading conceptual ideas of these
variations?

**Comment by Tomasz-W** [comment]
@Adamant36 

I've made a small visualisation comparing different green shades used in _osm-carto_ for certain plant types. I sort it by height, and I also added @imagico proposition from http://blog.imagico.de/more-on-pattern-use-in-maps/#comment-119715 and gardens/ plan nuresies of unspecyfied height.

![zielen](https://user-images.githubusercontent.com/25656654/44991635-00033c00-af95-11e8-8db4-8ecdf6bf25c3.jpg)

As` landuse=grass` and `landuse=meadow` has the same rendering and they may have very different height, I think https://github.com/gravitystorm/openstreetmap-carto/issues/3143 should be done first, and as `natural=heath` is very similar feature to grasses and grassland, I think we can use the same rendering then to avoid cluttering the map with next green shade.

**Comment by Adamant36** [comment]
@Tomasz-W, I was actually thinking the same thing. I thought id test it more anyway though. Scrub needs to be dealt with to. I was actually was going to see if you could make a color thing like that. So good idea. I guess the first thing to do is deal with #3143.

**Comment by jeisenbe** [comment]
"natural=heath is very similar feature to meadows and grasslands, I think we can use the same rendering for it"
I'm not sure. Heath is made of dwarf-shrubs, which are woody perineal plants. In subartic and alpine areas these may be stunted trees. 

It's quite different to walk through small bushes and shrubs, compared to a grassy field.

I also disagree with the idea that natural=grassland and landuse=meadow imply tall grasses. Grazed pasture is also tagged with landuse=meadow, and natural grasslands may be grazed (eg by elk and deer in North America, and various animals in Africa).

Also, grasses may never have time to grow tall in semi-arid  or arid climates. This idea of tall meadows and grasslands seems to be specific to a certain climate and landuse, and does not fit my experience in North or South America.

**Comment by Adamant36** [comment]
@jeisenbe, do you have any opinion on the difference between a heath and a scrub? They seem very similar to me. As they are both made up of woody plants, but they don't use the word heath in America where I map. So I might be wrong. If heath and scrub are essentially the same thing, maybe they should be rendered the same or very similar. Especially if they are never mapped next to each other and are essentially just regional terms for the same thing. 

In my opinion grassland and meadow are essentially the same thing. Except for the intentional usage of the grassy area for grazing or not by farmers. So I don't know how or if they should be rendered differently or not. Grazing land might be better categorized/rendered like an off shot of farmland. Since that's what it is in my mind, an extension of agriculture usage. Its often "managed" also. Although it wouldn't work to use the same color as farmland for it. Since it would confuse it with crop land. So I don't know. Maybe grass raised and managed for grazing/hay purposes could be considered a crop though. Having different crops rendered might help with that.

**Comment by jeisenbe** [comment]
As a Californian / Oregonian, I did not recognize “heath” either, but I
also had to look up “scrub”. In Siskiyou County we called scrublands
“brush”, and in Southern California there are extensive “Chaparral
scrublands.”

But we use the British terminology here for the tags, so heath is the
correct tag for vegetation dominated by “dwarf shrubs” (short woody-stemmed
plants)

@imagico menitioned previously that heathlands in Europe are often
anthropogenic; centuries of grazing by livestock or occasional mowing and
burning are what prevent heath from turning back to scrubland or woodland,
except in alpine, arctic and arid climates.

So in California you will find small areas of “heath” (dwarf woody plants)
above the tree line in the mountains, and perhaps in the semi-arid to arid
transition zones, and perhaps in rangelands that have been heavily grazed.

**Comment by dieterdreist** [comment]
2018-09-16 21:42 GMT+02:00 Adamant36 <notifications@github.com>:

> do you have any opinion on the difference between a heath and a scrub?
> They seem very similar to me. As they are both made up of woody plants, but
> they don't use the word heath in America where I map.
>


a scrub is dense, closed vegetation, usually not easily traversable, while
heath is open, scarse vegetation of certain types of grass and woody plants
that can live on these less fertile soils.



> So I might be wrong. If heath and scrub are essentially the same thing,
> maybe they should be rendered the same or very similar. Especially if they
> are never mapped next to each other and are essentially just regional terms
> for the same thing.
>


they are not

**Comment by jeisenbe** [comment]
Since we also want to change the scrub color, it may be necessary to change grassland color as well, to get a better heath color, which will also require changes to campsites, golf courses, leisure and parkland.  I believe heath with dde8ad and scrub with d1e0b4 should work, if the other changes are made.

Here are some tests in Northern Ireland with current Farmland eef0d5, forest/wood add19e, orchard/vineyard aedfa3. With changes to: **Grass/Meadow def6c0, heath dde8ad, scrub d1e0b4** and **park b5e3b5, pitch b8dabd**; leisure now includes golf and campsites.

**Rostrevor, Northern Ireland:**
Lots of heath, forest and a park
http://www.openstreetmap.org/#map=13/54.1072/-6.1147
Current rendering:
![greens-rostrevor-z13-before](https://user-images.githubusercontent.com/42757252/48753313-e38fb980-ecd0-11e8-8ab9-f0af9206ba60.png)
After (heath dde8ad):
![greens-rostrevor-ireland-after](https://user-images.githubusercontent.com/42757252/48753401-2f426300-ecd1-11e8-899c-e1d1875fc667.png)


**Newcastle, Northern Ireland:**
http://www.openstreetmap.org/#map=13/54.2141/-5.8593
Before z13
![greens-newcastle-z13-before](https://user-images.githubusercontent.com/42757252/48753413-3d907f00-ecd1-11e8-809b-6811f9d4f7c1.png)
After
![greens-newcastle-z13-after](https://user-images.githubusercontent.com/42757252/48753446-57ca5d00-ecd1-11e8-8b5d-23458df3b332.png)

**Comment by Tomasz-W** [comment]
@jeisenbe Please add a few tests renderings of `#d1e0b4` with 10% of this pattern https://gist.github.com/Tomasz-W/dc2b8fe24c9f50f60b49a40765835d5c

**Comment by jeisenbe** [comment]
I'm sorry, I don't understand what you mean by "10% of this pattern"? Do you want the pattern to be a light gray overlay on the d1e0b4 background, or do you want it to be a 10% darker version of d1e0b4? I may not know how to do this yet.

**Comment by Tomasz-W** [comment]
@jeisenbe I mean 10% opacity.

**Comment by jeisenbe** [comment]
The .png files for patterns normally have a slightly darker color than the background, in this style.

Unfortunately we are not able to use svg patterns directly due to limitations in mapnik+carto

Would you be able to remake this into a pixel-aligned png file using color 92ab77, and also try with a3b886?

However, I do think that heath should have a random pattern, as used for forest, scrub and other natural landcovers.

I've made some random png patterns to compare, with http://www.imagico.de/map/jsdotpattern.php
There's no perfect heath pattern there; we may need a new icon to use. But I've tried a couple options:

Round shrub pattern with 92ab77 for pattern (d1e0b4 background)
![bushes-92ab77-screenshot](https://user-images.githubusercontent.com/42757252/48810198-577ea000-ed6b-11e8-8fbd-07375371beb0.png)

Tall bushy pattern (a3b886)
![heath-grass-tall-screenshop](https://user-images.githubusercontent.com/42757252/48810258-af1d0b80-ed6b-11e8-9e8e-e4a734427e8f.png)

Short bushy pattern (92ab77)
![grass1-92ab77-screenshot](https://user-images.githubusercontent.com/42757252/48810275-cf4cca80-ed6b-11e8-847d-e7e362af81de.png)

These images were not pixel aligned; this makes it look a little more natural and random. But we could try pixel-aligning if we want a sharper, more regular look.

**Comment by jeisenbe** [comment]
But I still believe heath should get a different background color; as long as we keep it on the browner side of the spectrum it shouldn't be a problem.

Here are some possible patterns for heath with #aeb57c for the pattern and #d5dca1 for the background. It might be best to make an entirely new icon; I could also try a slightly darker color for the pattern:

Round pattern #aeb57c (on background #d5dca1 )
![heath-round-aeb57c-d5dca1-screen](https://user-images.githubusercontent.com/42757252/48810890-99f5ac00-ed6e-11e8-8b61-669730eb97be.png)

Tall bushy #aeb57c
![heath-tall-aeb57c-d5dca1-screen](https://user-images.githubusercontent.com/42757252/48810912-b2fe5d00-ed6e-11e8-8bd7-93953a3955f3.png)

Short bushy #aeb57c 
![heath-low-aeb57c-d5dca1-screen](https://user-images.githubusercontent.com/42757252/48811209-15a42880-ed70-11e8-9b41-9de9d6732afc.png)

Scrub-like #aeb57c  (It would be better to design a new svg pattern; I've just shrunk down the shrub icon to 0.8)
![scrub-smaller-aeb57c-d5dca1-screen](https://user-images.githubusercontent.com/42757252/48811236-37051480-ed70-11e8-91a3-1b40a06e78ae.png)

**Comment by jeisenbe** [comment]
Looking at the scrub patterns and forest, I believe that aeb57c may be too dark for the d5dca1 background. Here is b8c187 pattern on aeb57c:
![heath-tall-b8c187-d5dca1-screenshot](https://user-images.githubusercontent.com/42757252/48814854-ca921180-ed7f-11e8-9485-58e8249082dd.png)

Another option:
![scrub-heath-smaller-b8c187-d5dca1-screenshop](https://user-images.githubusercontent.com/42757252/48816124-455d2b80-ed84-11e8-95a1-5217baaa6244.png)

Ireland test areas:

**Kilbroney park**
http://www.openstreetmap.org/#map=14/54.1072/-6.1147
Current d6d99f:
![kilbroney-current](https://user-images.githubusercontent.com/42757252/48817506-2366a780-ed8a-11e8-9b80-20558704d41f.png)

Bushy b8c187 on d5dca1:
![kilbroney-heath-d5dca1-tall](https://user-images.githubusercontent.com/42757252/48817484-0f22aa80-ed8a-11e8-973a-bd8779e65b30.png)

Shrubby b8c187 on d5dca1:
![kilbroney-b8c187-d5dca1-smallscrub](https://user-images.githubusercontent.com/42757252/48817488-16e24f00-ed8a-11e8-8e54-9fb2e86b5cfb.png)

**Coastal Ireland**, 
heath mixed with grass, woods, and residential with houses, also next to a beach:
http://www.openstreetmap.org/#map=5/54.0578/-6.1949
Before d6d99f:
![forest-heath-grass-houses-before](https://user-images.githubusercontent.com/42757252/48817519-2e213c80-ed8a-11e8-9c47-ef08afcbeb48.png)

Bushy b8c187 on d5dca1:
![heath-tall-grass-forest-houses](https://user-images.githubusercontent.com/42757252/48817562-6c1e6080-ed8a-11e8-80d5-a9d747693371.png)

Shrubby b8c187 on d5dca1:
![forest-grass-houses-heath-b8c187-d5dca1-smallscrub](https://user-images.githubusercontent.com/42757252/48817569-717bab00-ed8a-11e8-8c81-a32de3775f89.png)

**Glen River, near Newcastle**
http://www.openstreetmap.org/#map=14/54.1991/-5.9125
Before d6d99f:
![glenriver-before](https://user-images.githubusercontent.com/42757252/48817591-8bb58900-ed8a-11e8-8496-c58905b79fff.png)

Bushy b8c187 on d5dca1:
![glenriver-heathtall-b8c187-d5dca1](https://user-images.githubusercontent.com/42757252/48817600-91ab6a00-ed8a-11e8-9010-e439dcb017e1.png)

Shrubby b8c187 on d5dca1:
![glenriver-smallscrub-b8c187-d5dca1](https://user-images.githubusercontent.com/42757252/48817608-9cfe9580-ed8a-11e8-8f88-c94ad79ad7e8.png)

**Garry Bog**
Bogs use the heath color as the background, although the pattern is different (so this is unchanged). 
Here is a comparison of the background before and after:
Before d6d99f:
![garrybog-before](https://user-images.githubusercontent.com/42757252/48817640-c4556280-ed8a-11e8-926f-839367123eed.png)

After d5dca1:
![bog-heath-d5dca1](https://user-images.githubusercontent.com/42757252/48817647-c7505300-ed8a-11e8-8616-ba4ec677394b.png)

It's a subtle change, but perhaps it is enough, with the addition of a pattern at zoom 14 and up. 

**New Scrub plus New Heath** - test area
Scrubby heath d5dca1 with scrub d1e0b4
![new-scrub-new-heath-scrubby](https://user-images.githubusercontent.com/42757252/48818720-d08fee80-ed8f-11e8-95d6-cf6296dc0658.png)

Bushy heath with new scrub
![bushy-heath-new-scrub](https://user-images.githubusercontent.com/42757252/48818717-c5d55980-ed8f-11e8-928e-a06a2e0f14ee.png)

Probably the second symbol is better, or perhaps we should try something different for the symbol.

**Comment by jeisenbe** [comment]
Here I've changed the heath background color to **dde5a9,** slightly lighter, and the pattern is also a little lighter to match: **c0c98e**. This is about as much as it can be lightened without getting problems with contrast against grass color, unless we change grass/meadow as well.

_Looks ok next to beach, grass, wood and residential:_
![coast-bushyheath-c0c98e-dde5a9](https://user-images.githubusercontent.com/42757252/48821766-f2906d80-ed9d-11e8-8986-0a8a95cc3b8b.png)

_Ok with streams, rivers, paths and tracks:_
![glenriver-bushyheath-c0c98e-dde5a9](https://user-images.githubusercontent.com/42757252/48821767-f3290400-ed9d-11e8-8e7f-b9f3069c4dc0.png)

![kilbroney-bushyheath-c0c98e-dde5a9](https://user-images.githubusercontent.com/42757252/48821775-f7552180-ed9d-11e8-91b8-8b7885adea84.png)

OK contrast with the new scrub color and pattern, also with parks and farmland:
![test1-bushyheath-and-newscrub-c0c98e-dde5a9](https://user-images.githubusercontent.com/42757252/48822058-f96bb000-ed9e-11e8-89a4-31880ea6b751.png)

![test2-bushyheath-and-newscrub-c0c98e-dde5a9](https://user-images.githubusercontent.com/42757252/48822060-fbce0a00-ed9e-11e8-8efc-1e0cc1a38d49.png)

Color comparisons: http://davidjohnstone.net/pages/lch-lab-colour-gradient-picker#eef0d5,dde5a9,c0c98e,def6c0,dde5a9,c0c98e,d1e0b4,dde5a9,c0c98e,add19e,dde5a9

The png files are available at https://github.com/jeisenbe/openstreetmap-carto/tree/heath-scrub/symbols/

**Comment by Adamant36** [comment]
Thanks. I think it looked slightly better when it was darker like in the last examples. It stands out a little to much in the lighter examples for my taste. Other then that I think it looks good and is good to go. I say you do a few comparison of the darker and lighter colors in the same places. Then leave it for a few days so other people can have time to provide feedback. After that, it should be PR worthy.

**Comment by Tomasz-W** [comment]
> Would you be able to remake this into a pixel-aligned png file

@jeisenbe Here you are:
png: https://wiki.openstreetmap.org/wiki/File:Meadow_pattern_proposal.png
svg: https://gist.github.com/Tomasz-W/dc2b8fe24c9f50f60b49a40765835d5c

**Comment by jeisenbe** [comment]
@Adamant36 here are some more in the Azores islands, which have a few areas with very detailed mapping of scrub, heath, grass and woods, along with some streams and intermittent streams:

Azores
http://www.openstreetmap.org/#map=15/38.5627/-28.6741
z17 Before
![z17-flamengos-before](https://user-images.githubusercontent.com/42757252/48826120-d811c080-edac-11e8-99b7-52a4bcd06a28.png)
After
![z17-flamengos-scrub-ceddb0-heath-dde5a9](https://user-images.githubusercontent.com/42757252/48826121-d942ed80-edac-11e8-88c3-2530043fff99.png)

Caldeira Before
![caldeira-before](https://user-images.githubusercontent.com/42757252/48826132-dea03800-edac-11e8-8c4b-1fc83a7da8ba.png)
After
![caldeira-scrub-ceddb0-heath-dde5a9](https://user-images.githubusercontent.com/42757252/48826137-e364ec00-edac-11e8-94cc-3ef885c8dc35.png)

Flamengos Before
![ribeira-flamengos-before](https://user-images.githubusercontent.com/42757252/48826153-f081db00-edac-11e8-831d-a6209e6ef0d3.png)
After
![flamengos-scrub-ceddb0-heath-dde5a9](https://user-images.githubusercontent.com/42757252/48826164-f7105280-edac-11e8-97ba-bb93c2f6b2ca.png)

Casa-do-Cantoneiro Before
![cantoneiro-before](https://user-images.githubusercontent.com/42757252/48826190-042d4180-edad-11e8-80af-f3cfd732b8c2.png)
After
![casa-do-cantoneiro-scrub-ceddb0-heath-dde5a9](https://user-images.githubusercontent.com/42757252/48826197-07283200-edad-11e8-98e5-7215cb1a725b.png)

Estrada Before
![estrada-before](https://user-images.githubusercontent.com/42757252/48826213-0c857c80-edad-11e8-9101-a983937036ec.png)
After
![estrada-after-scrub-heath](https://user-images.githubusercontent.com/42757252/48826217-10190380-edad-11e8-8702-841330eeb9c5.png)

And here is a big area of heath with intermittent streams in Northern Australia:
After
![heath-intermittent-stream-australia](https://user-images.githubusercontent.com/42757252/48826253-21faa680-edad-11e8-8a64-b93fbb01dd57.png)

(The scrub color in these pictures was ceddb0, just 1 or 2% darker than d1e0b4, but heath is still dde5a9)

**Comment by Adamant36** [comment]
Thanks again for the tests. I still think the darker ones look best, but its a superficial difference and the lighter colors for both are way better then what they currently are. 

Side by side of light and dark scrub done in paint (kind of crap, but whatever).
![darker and lighter scrub](https://user-images.githubusercontent.com/30259065/48834851-97c22c00-ed33-11e8-8de6-ffb6702941a8.png)

**Comment by Tomasz-W** [comment]
I think that using this pattern for heath areas is a bad idea - we use "much spaced" (sorry I don't know how to name it) icon-based patterns rather for higher landcover (e.g. forests, scrubs), and for lower landcover we use no-pattern or simple geometrical-based dense patterns (e.g. wetlands, gardens). Mixing this 2 ways might be confusing for map users, and we should keep things intuitive. That's the reason why I think that reusing rotated wetland pattern with 0.1 opacity would work here the best. Notice also that with very different patterns for scrub and heath. there is a possilibily that we would can use the same colour as their background, because of the different patterns they would be distinguished enough. I'm looking forward to see some test renderings of https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-440558091 on `#d1e0b4` to make sure if is it a good idea or not.

**Comment by Adamant36** [comment]
I suggest we forgo the pattern on heath for now and just focus on the color. Since its the purpose of this issue and otherwise it convolutes things to much. Once the PR for the heath color change is merge we can start a new issue for adding a pattern if need be. Its better not to mix issues though. 

Also, @jeisenbe, if you don't mind id like to do the PR for this and the color change on scrub. Since you already have 4 PRs your currently working on. It would be good if you focused on getting those merged. Plus, I feel like its better/will go quicker in general if we can delegate tasks to each other and share the PRs when need be. Otherwise things risk getting spread to thin or derailing for other reasons. Especially since we only have one maintainer doing code review/PR merging at this point.

**Comment by Tomasz-W** [comment]
> I suggest we forgo the pattern on heath for now and just focus on the color. 

Notice that some dense pattern overlay may change colour perception, so I think that testing different options of colour + pattern at one time would be a better way of work on this ;)

**Comment by Adamant36** [comment]
Normally id agree with you, but it doesnt seem like we are even decided on if heath will use a pattern in the place. While we already pretty much have a color picked out. Plus, a pattern or not shouldnt decide the color. The color should stand on its own as good or not. The pattern is ultimately an unnecessary, but still good, addition. Whereas, it has to have a color. It was fine all this time without a pattern. That's not to say we shouldn't add one though. But assuming its what heath needs, while throwing color to the wayside and without hearing the opinions of others, is just bad policy. That's just my opinion though and I'm not saying we are doing that, but I havent seen anyone else besides you, me, and Jeisenbe even have an opinion on it yet. In support or not. Whereas, I know most in the community would be OK with a color change. Not to mention I seem to remember being the one that was working on it originally. Not that it matters that much though. At least though to me, there's both pros and cons of patterns and we should consider if this a good situation for it before jump the gun on design etc. There's been zero disscussion of its even something we want in the first place, which there would be if it was handled in a seperate issue. 

@kocio-pl, what do think about adding a pattern to heath versus just changing the color for now?

**Comment by kocio-pl** [comment]
Sorry, I do not follow this discussion in details, because it's very broad and complicated issue (since this is related to other propositions of changing greens) and I have limited time for OSM lately. 

I try to focus on the merging queue and release, since this critical part now.

**Comment by Adamant36** [comment]
@kocio-pl, OK. Fair enough.

**Comment by Tomasz-W** [comment]
I think that proposed colour is too intensive/ too dark - as generally it turns out that the darker the green fill is, the higher plants it includes, proposed heath share breaks it imo. I'll propose some green shade for it later.

**Comment by Tomasz-W** [comment]
`#d7e5ac` for heath (with `#d1e0b4` for scrub) proposal:

![scrub1](https://user-images.githubusercontent.com/25656654/48867340-2b4c3900-edd6-11e8-9668-32cd5fc89dc4.jpg)
![scrub2](https://user-images.githubusercontent.com/25656654/48867343-2be4cf80-edd6-11e8-9b5e-7ab674c0ae86.jpg)
![scrub3](https://user-images.githubusercontent.com/25656654/48867345-2be4cf80-edd6-11e8-8066-73a77d7af201.jpg)

**Comment by jeisenbe** [comment]
Sorry I haven't responded; for some reason this thread was sent to spam in my email.

I've tried a regular dash pattern for heath as @Tomasz-W suggested, but it looks too man-made to me; it's basically the same pattern as vineyard with a browner background color, which suggests some sort of agricultural area IMO. Here heath and scrub have the same background and pattern color:

![flamengos-scrub-heath-dashes](https://user-images.githubusercontent.com/42757252/48874773-510a2a00-ee38-11e8-91eb-e8b3f588c5e5.png)
![estrada-scrub-heath-dashes](https://user-images.githubusercontent.com/42757252/48874774-523b5700-ee38-11e8-9088-37e2ffbfcd4a.png)
![ribeira-scrub-heath-dashes](https://user-images.githubusercontent.com/42757252/48874775-52d3ed80-ee38-11e8-97bb-4ccf4348e72a.png)

It would work a little better with a random pattern, but I would still reserve this kind of pattern for grass/meadow

**Comment by Adamant36** [comment]
I agree about the pattern. It looks to man made. As far as the new colors, I prefer the older ones. Scrub was fine darker and I don't feel like grass needs a change. Unless there's actually a good reason to. Otherwise, its easy for a change to one thing to justify a change to another one, etc etc. Id note to that heath/scrub is a "natural" tag. Whereas grass is a "landuse" one. Therefore, they will inherently have different tones and "clash" slightly. As they should.

**Comment by jeisenbe** [comment]
_Re: "I prefer the older one",_ 
Which colors do you mean, @Adamant36 ? 

Do you mean these colors (ceddb0 scrub and dde5a9 heath): https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-440568617 

Or d1e0b4 scrub and d5dca1 heath: https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-440524240

I think the heath pattern in the most recent tests was a little too dark. Here is a new pattern that is closer to the background color (c0cc96 instead of b5c498), and I've also made the pattern less dense, so it matches better with scrub. (Also, if we make the heath pattern more dense, mappers may start tagging areas of scrub as natural=heath where the bushes are close together, but this is incorrect.)

**Heath #d7e5ac and scrub #ceddb0:**
![export-11](https://user-images.githubusercontent.com/42757252/48877134-2f636f80-ee45-11e8-918a-626b5baeb89a.png)

![export-12](https://user-images.githubusercontent.com/42757252/48877150-486c2080-ee45-11e8-9f34-5c3a912f8fa2.png)

![export-13](https://user-images.githubusercontent.com/42757252/48877152-4a35e400-ee45-11e8-818b-45e7daa773be.png)

![export-15](https://user-images.githubusercontent.com/42757252/48877154-4b671100-ee45-11e8-9643-74e5f2d84a59.png)

![export-16](https://user-images.githubusercontent.com/42757252/48877155-4c983e00-ee45-11e8-80d0-165170c9effc.png)

**Comment by Adamant36** [comment]
d1e0b4 scrub and d5dca1 heath. I think. I'm starting to lose track since you keep changing multiple things things in the same test. I.E "can I see test of scrub at d1e0b4 around different things?", "Sure here's a test scrub at #ceddb0 darkened by %15 and a heath at c0cc96 with a new pattern even though know one commented on what they thought about the last one." etc etc. 

As it is, I think d1e0b4 is good for scrub and I like the bushy pattern for heath compared for the lines. As far as the heath color goes, I'm not really sure at this point. Since you know my opinion and I can't keep up anymore I think I'm going to dodge out of this one for a while until you and @Tomasz-W figure out the details. I have faith in you guys to figure it out ;)

**Comment by jeisenbe** [comment]
Here we can compare the different versions:

0. Before (current master): darkest
![](https://user-images.githubusercontent.com/42757252/48817519-2e213c80-ed8a-11e8-9c47-ef08afcbeb48.png)

1. Bushy b8c187 on d5dca1: close to current color (scrub d1e0b4)
![](https://user-images.githubusercontent.com/42757252/48817562-6c1e6080-ed8a-11e8-80d5-a9d747693371.png)

2. Bushy c0c98e on dde5a9, slightly lighter (scrub d1e0b4)
![](https://user-images.githubusercontent.com/42757252/48821766-f2906d80-ed9d-11e8-8986-0a8a95cc3b8b.png)

3. Newest pattern with heath d7e5ac (Tomasz-W's suggestion; slightly lighter and greenr) with scrub ceddb0:
![coastal-d7e5ac-ceddb0](https://user-images.githubusercontent.com/42757252/48877704-bf0a1d80-ee47-11e8-88e3-aed543fc48a9.png)

Re: "grass"; recall that the same color is used for pastures, meadows, landuse=grass and natural=grassland, as well as village_green and common, and it's used as the background color for garden. It's common to find heath and scrub next to meadow, pasture or natural=grassland; the colors should look good next to each other.

**Comment by jeisenbe** [comment]
Comparison of the different versions of heath vs grass, forest and residential:
http://www.openstreetmap.org/#map=5/54.0578/-6.1949 coastal Northern Ireland

0. Before (current master): darkest
![](https://user-images.githubusercontent.com/42757252/48817519-2e213c80-ed8a-11e8-9c47-ef08afcbeb48.png)

1. Bushy b8c187 on d5dca1: close to current color; @Adamant36 prefers this one, he thinks.
![](https://user-images.githubusercontent.com/42757252/48817562-6c1e6080-ed8a-11e8-80d5-a9d747693371.png)

2. Bushy c0c98e on dde5a9, slightly lighter
![](https://user-images.githubusercontent.com/42757252/48821766-f2906d80-ed9d-11e8-8986-0a8a95cc3b8b.png)

3. Newest pattern with heath d7e5ac; slightly lighter and greener - @Tomasz-W's new suggestion
![coastal-d7e5ac-ceddb0](https://user-images.githubusercontent.com/42757252/48877704-bf0a1d80-ee47-11e8-88e3-aed543fc48a9.png)

Re: "grass"; recall that the same color is used for pastures, meadows, landuse=grass and natural=grassland, as well as village_green and common, and it's used as the background color for garden. It's common to find heath and scrub next to meadow, pasture or natural=grassland; the colors should look good next to each other.

**Comment by Tomasz-W** [comment]
@jeisenbe Pattern showed in https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-440870962 is based on some other file than the one I provided in https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-440558091, so please test the second one on:

* `#d1e0b4` (scrub colour)
* `#d7e5ac`
* `#dde5a9`
* `#dce5ac` (a little less saturated version of the #dde5a9)

To have a simple and connected system, I would use this pattern also for meadows (on grass background) and grasslands (on a `#e4efca` from https://github.com/gravitystorm/openstreetmap-carto/issues/3411#issuecomment-424900037)

**Comment by jeisenbe** [comment]
I disagree with the use of a regular pattern for heath (or grassland for
that matter), and I do not think we should use the same pattern for
different types of vegetation.

**Comment by Adamant36** [comment]
@jeisenbe, you don't think the three different patterns clash/look off/to busy
when next to each other due to not being aligned etc (for instance, in this picture)?

![48822058-f96bb000-ed9e-11e8-89a4-31880ea6b751](https://user-images.githubusercontent.com/30259065/48897355-b93c1a00-edfe-11e8-9989-365dacaf22b5.png)

**Comment by Tomasz-W** [comment]
@jeisenbe Pattern which I provided above is irregular (you took some another one for test, propably by mistake), so again: please test https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-440558091 on a colours proposed in https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-440943178.

**Comment by jeisenbe** [comment]
Re: _Three different patterns._

@Adamant36, that example was made up for testing. However, there certainly are areas with multiple landcover patterns next to each other, even now. The patterns are needed because we render too many types of landcover for them to be understandable otherwise, and the pattern is random because a regular pattern looks unnatural.

For example, this area in Wales has marsh, woods and scrub, which all have random patterns:
![scrub-marsh-grass](https://user-images.githubusercontent.com/42757252/48903296-75005680-ee9e-11e8-8172-ac00e26370a2.png)

**Comment by Adamant36** [comment]
@jeisenbe, heath hasn't had a pattern this whole time and people knew what it was. Plus, this issue didn't have anything to do with a pattern originally and know one has asked for it have one. patterns are useful in situations where the color of things are similar, like dog parks versus regular parks. The color of heath, including the new one, is different enough though that it is easy to tell what it is compared to the other things around it. 

As I said above and on your other issues, I wouldn't add complexity just for the sake of adding complexity. Otherwise, it over complicates things pointlessly. I also wouldn't assume mappers need things when there's no evidence for it. If you want a pattern added because you think it should be there fine, but there's zero evidence it would help people tell what heath is better. Its not like someone start a new issue if its there's a need for a pattern later. Otherwise, maybe we should put patterns on retail, commercial, and residential areas etc etc. Because you never know right? 

Anyway, maybe the example I provided of the chaotic patterns was an outlier. The one you provided looks better. Maybe its just because the heath color is still bright and stands out to much. Its to much on the yellow side. 

If we are going with a pattern and heath/scrub are essentially the same thing is there a reason they can't both have the scrub color (or the scrub color slightly darkened? I.E. like park and dog park. Otherwise, the pattern its just pointless extra noise. Since they are obviously different colors as it currently is. (btw, I knew park and dog park aren't exactly the same color, but they are close enough. Maybe a better analogy would be dog park and playground?).

**Comment by jeisenbe** [comment]
Sorry, @Tomasz-W, I had remembered a regular pattern suggested for meadow in another issue, and for some reason I thought that was what you were suggesting here. I see that it is similar but random.

Mapnik will not let us adjust the color of a png, as far as I know, so I would have to create a new png for testing with each of the 4 background colors you suggested; generally I have been darkening the background color about 10% to get the pattern color. 

That's why I asked for a png that already had the right color, eg c0c98e for testing with d7e5ac, or b5c498 for d1e0b4, etc. I don't have photoshop or other good image editing software.

I've been creating patterns at http://www.imagico.de/map/jsdotpattern.php - you can make a random pattern similar to that by using the dash, shrinking it by 50%, and then rotating 90 degrees after exporting and converting to png. 

Here's a couple examples with a similar pattern (color #c0c98e) on heath (#d7e5ac):

![heath-dashes1](https://user-images.githubusercontent.com/42757252/48906294-56528d80-eea7-11e8-97ce-60e11bdde05f.png)

![heath-dashes2](https://user-images.githubusercontent.com/42757252/48906301-581c5100-eea7-11e8-8bb2-329d439c7ec9.png)

Still makes me think of some type of cropland

**Comment by Adamant36** [comment]
Btw, read comment here and the ones around about having to justify a PR and how your endorsing it by doing it https://github.com/gravitystorm/openstreetmap-carto/pull/3402#issuecomment-422366396. If you can't provide a compelling reason for why your changing something or why exactly its needed, it probably isn't a good change. Just so you know, I'm not trying to be whatever about. I just want things to go smoothly for our time not to be wasted on issues that aren't worth doing/won't get merged because they can't be justified.

**Comment by Tomasz-W** [comment]
@jeisenbe Thanks, but there is still something wrong - as this is rotated wetland pattern, it should has the same symbol-density density level, but somehow it hasn't. Can you investigate it and make pattern denser? I still would like to rate its effect on colours from https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-440943178.

**Comment by jeisenbe** [comment]
You are correct, it wasn’t the same pattern. As I mentioned, I don’t have
photoshop or other image editing software, so I generated a new pattern at
the link. But I think it gives some idea of what it would look like.

I think shorter dashes would look better, for grassland, but I don’t think
it will work for heath.

**Comment by Adamant36** [comment]
d7e5ac isn't going to work as a heath color because its to light and blends in with the other green. So heath needs to be darker. 
![d7e5ac closeup](https://user-images.githubusercontent.com/30259065/48910665-66298d80-ee25-11e8-95f0-29c520ccd0c3.png)
I kind of like heath at scrub (d1e0b4) darkened by 2% or 5%. 
2% 
![scrub darken 2](https://user-images.githubusercontent.com/30259065/48911119-bce39700-ee26-11e8-9d8e-2800bce5ac81.png)
5%
![scrub darken 5](https://user-images.githubusercontent.com/30259065/48911129-cb31b300-ee26-11e8-9a19-921e9349082d.png)

**Comment by jeisenbe** [comment]
Great testing location! I like seeing residential,  farmland, grass, vineyards, heath, scrub, and forest in one place. Could you share the link to this place?

What colors are these? I'm guessing that d1e0b4 darkened by 5% is something like c3d2a5?

The scrub color should be darker than heath; consider that woodland is one of the darkest landcovers currently, and farmland and grass are light. So higher vegetation is darker. 

We could try using c3d2a5 for scrub and d1e0b4 for heath, then. But there colors are fairly similar to each other:
![export-14](https://user-images.githubusercontent.com/42757252/48925303-a3654c80-ef05-11e8-93d0-ee6f129fc685.png)
The scrub and heath are distinct when they are right next to each other and have a pattern.
![export-17](https://user-images.githubusercontent.com/42757252/48925310-b710b300-ef05-11e8-9311-16976cf04255.png)
Could you tell that this was a patch of scrub next to forest in the last image? Can you tell if the small areas at the edge of the image (at top left and top right) are scrub or heath without a pattern?
![export-18](https://user-images.githubusercontent.com/42757252/48925307-b1b36880-ef05-11e8-82f7-747d0e2b92a7.png)

**Comment by jeisenbe** [comment]
Here are the same areas with c8d7ab
![3a-d1e0b4-c8d7ab](https://user-images.githubusercontent.com/42757252/48926076-db708d80-ef0d-11e8-9637-c306d5e07b84.png)
 (3% darker) for scrub and d1e0b4 for heath. Now the scrub is more distinct from the forest, but the scrub and heath are very similar without a pattern.
![1-c8d7ab](https://user-images.githubusercontent.com/42757252/48925619-17552400-ef09-11e8-87a4-bdf866cbee7d.png)
I can't easily distinguish heath and scrub at zoom 13:
![2-c8d7ab](https://user-images.githubusercontent.com/42757252/48925622-1c19d800-ef09-11e8-8b44-692922d04363.png)
But now the scrub and wood are different enough:
![3-c8d7ab](https://user-images.githubusercontent.com/42757252/48925623-1cb26e80-ef09-11e8-87f1-e36dced20f24.png)

Next to grass:
![3a-d1e0b4-c8d7ab](https://user-images.githubusercontent.com/42757252/48926100-2ee2db80-ef0e-11e8-83e1-a924cdfc8d07.png)

-----------------------------------------

So if we are going in this direction, I would suggest making heath a slightly more yellow shade than d1e0b4. This will also help keep bogs looking right. Tomasz-W suggested dce5ac above, but this is also a little too close to grass color. If we make d1e0b4 more yellow we get d9deb0, so I'll try that for heath.

Here's d9deb0  for heath (d1e0b4 shifted toward yellow), and c8d7ab for scrub (3% darker d1e0b4):
![1-d9deb0-heath-c8d7ab-scrub](https://user-images.githubusercontent.com/42757252/48925790-12916f80-ef0b-11e8-8eb7-e56e397c8c85.png)
![2a-d9deb0-heath-c8d7ab-scrub](https://user-images.githubusercontent.com/42757252/48925791-14f3c80-ef0b-11e8-8d75-e961cbba7525.png)
![2b-1-d9deb0-heath-c8d7ab-scrub](https://user-images.githubusercontent.com/42757252/48925792-158c6000-ef0b-11e8-9716-e09f360c7aa3.png)

Here is another area, with meadow or grass next to the scrub and heath, plus some streams:
![3a-grass-d9deb0-heath-c8d7ab-scrub](https://user-images.githubusercontent.com/42757252/48925941-c21b1180-ef0c-11e8-8b60-a7b3826ff512.png)
![3b-grass-d9deb0-heath-c8d7ab-scrub](https://user-images.githubusercontent.com/42757252/48925942-c3e4d500-ef0c-11e8-8d38-87809b1499ca.png)

(I've also edited the previous comment to add this test area, in case you are reading on email https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-441138956)

**Comment by jeisenbe** [comment]
OK, @Tomasz-W, here is one of your requests. I didn't manage to get the two patterns to match the same density, but I've added a random vertical dash pattern for both grassland and heath.

The grassland pattern is  #b3d097 (on the current grass color background, cdebb0)
The heath dash pattern #c0c98e is with a background color of #dde5a9 for heath.
Scrub background is #d1e0b4:

Here's an area in Wales with grassland next to heath, with some forest, scrub and wetland nearby. The similar patterns make the two areas look more alike, which is a problem:
![grassland-heath](https://user-images.githubusercontent.com/42757252/48930570-c65b2500-ef34-11e8-9dd8-ae8c4191a73d.png)
Heath with scrub and forest:
![export-19](https://user-images.githubusercontent.com/42757252/48930578-d6730480-ef34-11e8-84c7-536af09e3c18.png)
Next to meadow, scrub and woods (I did not put a pattern on meadow)
![export-14](https://user-images.githubusercontent.com/42757252/48930600-f6a2c380-ef34-11e8-887f-2e5cff5111cc.png)
z13; no patterns, just background colors: (except wood, which still gets a pattern at this level)
![export-20](https://user-images.githubusercontent.com/42757252/48930586-e2f75d00-ef34-11e8-984b-9d6a6f2239f5.png)
![export-17](https://user-images.githubusercontent.com/42757252/48930610-0e7a4780-ef35-11e8-9d5a-41bcc5f85685.png)

**Comment by jeisenbe** [comment]
Re: pattern with short vertical lines.

I agree that this pattern doesn’t work well. I would prefer more widely
spaced icons that look like shorter bushes.

Re: color of scrub. As I mentioned above, it starts looking similar to
woodland/forest, especially on z13 and lower, if the scrub color is too
dark.

One option would be to make heath a bit lighter so that the scrub looks
good, and also lighten grass so that heath is not too similar.

Current grass/meadow is much darker than farmland, so we can lighten it
without problems as long as we change camp sites (these look almost the
same as grass already).

Is it worth opening up that can of worms?

**Comment by jeisenbe** [comment]
Tomasz Wójcik wrote:

> I like an effect of pattern abov, ... , It's also a simplest way to make a
> difference between lowest-level grass (no-pattern), higher heath/
> grassland/ meadow (dashes pattern on different backgrounds) ...
>

While natural=grassland may grow high in areas of regular rainfall like
northern Europe, grasslands are very common in semiarid areas like the
western USA, Australia, northern Africa, Central Asia etc, and there the
grass is quite short. Most natural grasslands are gazed by wild (or
domestic) animals, eg Bison and Elk in the USA, which also keeps the grass
short for much of the year.

**Comment by Adamant36** [comment]
> Re: color of scrub. As I mentioned above, it starts looking similar to
woodland/forest, especially on z13 and lower, if the scrub color is too
dark.

I don't agree with that. The olive color (#d1e0b4) doesn't look anything like woodland/forest. As far as changing grass/meadow. Grass is for managed grass, like lawns. Which is naturally a lot darker. So its fine the color it is. Meadow could be lightened closer to farmland though. As I've suggested more then once already.

**Comment by Tomasz-W** [comment]
After some Photoshop tests, I like an idea of `#d1e0b4` for heath and 3% darken  `#d1e0b4` for scrub :) Patterns are another thing.

**Comment by jeisenbe** [comment]
Remember that landuse=meadow is also approved to be used for pasture; in
fact there are many areas where most of the places tagged meadow are grazed
full-time as pasture.

Eg all of the areas tagged landuse=meadow in my hometown in Northern
California are pasture, most are irrigated in the summer, and grazed by
cattle or horses most of the year, so the grass is green and short.

**Comment by jeisenbe** [comment]
> After some Photoshop tests, I like an idea of `#d1e0b4` for heath and 3% darken `#d1e0b4` for scrub

Do you mean c8d7ab for scrub, like in the first few photos in this comment: https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-441157955

I think this color will work well for scrub, but I'll try some tests with intermittent streams on the appropriate thread. Let's try to get that settled first, unless we want to do grass/heath/scrub all in one PR.

**Comment by Adamant36** [comment]
in the code it would just be @heath: #d1e0b4; 

Then @scrub: darken(@heath, 3%);

The hex value doesn't really matter when its done that way. 

I think we can do scrub/heath in their own PR since they go together and then deal with grass separately on its own. Although, I think its fine with the current color (unless you mean grassland/meadows. Those could be changed, but it should still be in a different PR then scrub/heath).

**Comment by jeisenbe** [comment]
It’s better to have independent colors for separate features when possible,
especially since the patterns are png files and we have to generate it with
the right colors.

**Comment by jeisenbe** [comment]
**Re: Patterns**
The Ordnance Survey is one of the examples of maps that shows a pattern for heath. I find the legend a little confusing, but one of these patterns is used for heath on the current maps (I believe the one on top; the others are bracken and rough grass?):

![bracken-heath-rough-grass](https://user-images.githubusercontent.com/42757252/48963709-f0b3ed80-efdb-11e8-8f8f-ea08f1e0cee4.png)

The old OS maps included this legend. The resolution of this image is poor, but you can see "brushwood" (scrub) in the upper middle, and "Furze" in the lower middle for a rounder pattern. Furze is the same as gorse, or broom; a type of heath. 
![old-os-legend-brushwood-heath](https://user-images.githubusercontent.com/42757252/48963718-032e2700-efdc-11e8-9ae2-5012a22b91fe.png)

Perhaps we can try a symbol for heath that is similar to one of these.

**Here are some examples of heath from outside of Europe:**

http://www.anbg.gov.au/photo/vegetation/heathlands.html
![](http://www.anbg.gov.au/images/photo_cd/508_FAGG/DSC_0948_2.jpg)

![](http://www.anbg.gov.au/images/photo_cd/649_FAGG/R0011663_2.jpg)

![](https://upload.wikimedia.org/wikipedia/commons/thumb/5/51/Die_back_valley_gnangarra.jpg/800px-Die_back_valley_gnangarra.jpg)

Diagrams of Australian heath structure:
![](http://www.anbg.gov.au/photo/vegetation/structure/closed-heath-structure.jpg)

![](http://www.anbg.gov.au/photo/vegetation/structure/open-heath-structure.jpg)

Compare to Australian grasslands, the "Brunette Downs" from the [Wikipedia Grasslands page](https://en.wikipedia.org/wiki/Grassland):
https://upload.wikimedia.org/wikipedia/commons/thumb/8/83/Brunette_Downs.jpg/800px-Brunette_Downs.jpg

South African Heath:
https://upload.wikimedia.org/wikipedia/commons/8/85/Fynbos.jpg

Alpine/Montane heath
https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Alpine_Heath-like_Daisy_%283193520823%29.jpg/800px-Alpine_Heath-like_Daisy_%283193520823%29.jpg

**European examples** for reference, from [Wikipedia Heath page](https://en.wikipedia.org/wiki/Heath
):
Heath dominated by yellow flowering Broom aka Gorse aka Furze in England:
![](https://upload.wikimedia.org/wikipedia/commons/8/8a/Heath.jpg)

Purple heather in Germany:
https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Lüneburger_Heide_109.jpg/800px-Lüneburger_Heide_109.jpg

**Comment by jeisenbe** [comment]
Having said all that, here are some real-world test images with a horizontal dash pattern for grasslands, similar to what was used in the OS map legend. Perhaps this looks better than the vertical dash pattern.

There are also some marshes in these images; even though the pattern is similar in shape, the different color is distinct enough, I think. 

z14
![pasir-putih-shorter-dashes](https://user-images.githubusercontent.com/42757252/48965038-94ab9200-eff8-11e8-9067-cb43c1caf3f9.png)

z15


z16
![Uploading Wesaput-z16-shorter-dashes.png…]()

Meadow/pasture are not given a pattern, for contrast. These test areas were mapped by myself based on survey plus aerial imagery; meadow represents land that is transitioning from farmland back to scrub and the scrub is mainly transitional to forest (or sometimes on poor soil), while the "natural" grasslands are rocky hills where the grass is maintained by occasional burning, rather than grazing. Farmland is mainly sweet potato fields, with some rice, taro and other vegetables.

**Comment by Tomasz-W** [comment]
@jeisenbe Can you make a test rendering of `#d1e0b4` with a vertical pattern version from https://github.com/gravitystorm/openstreetmap-carto/issues/3143#issuecomment-441346233?

**Comment by jeisenbe** [comment]
**Re "Can you make a test rendering of #d1e0b4 with a vertical version of a pattern from #3143?"**

I believe d1e0b4 is too close to d1e0b4 (it's only 3% lighter, and otherwise identical). 

Here is a comparison of Snowdonia National Park in Wales, from z10 to z16, with the new scrub color (#c8d7ab) as the only change, or with the new scrub color (#c8d7ab) and with #d1e0b4 for heath. I've also shown a third test with #d1e0b4 for scrub and #d9deb0 for heath. 

The color #d9deb0 has the same lightness and chroma as #d1e0b4, but is less green (hue is 113, instead of 122), so it looks more similar to the earlier test colors of heath, and not too similar to the new scrub color.  It also looks less similar to grass, because grass is also more green (hue 128)

z16 Scrub c8d7ab
![z16-snowdonia-scrub-only](https://user-images.githubusercontent.com/42757252/48975148-0691e300-f0ae-11e8-82f0-b0c7b41b2437.png)

z16 Heath d1e0b4, scrub c8d7ab
![z16-new-scrub-heath-d1e0b4](https://user-images.githubusercontent.com/42757252/48975150-098cd380-f0ae-11e8-989c-97c0f4eb1f67.png)

z16 Heath d9deb0, scrub c8d7ab
![z16-new-scrub-heath-d9deb0](https://user-images.githubusercontent.com/42757252/48975159-2aedbf80-f0ae-11e8-90b9-a29f9578fd06.png)

z14 Scrub c8d7ab
![z14-snowdonia-scrub-only](https://user-images.githubusercontent.com/42757252/48975160-2e814680-f0ae-11e8-9432-456fc7daa5f5.png)

z14 Heath d1e0b4, scrub c8d7ab
![z14-new-scrub-heath-d1e0b4](https://user-images.githubusercontent.com/42757252/48975162-33de9100-f0ae-11e8-9a89-886c0e6ffeef.png)

z14 Heath d9deb0, scrub c8d7ab
![z14-new-scrub-heath-d9deb0](https://user-images.githubusercontent.com/42757252/48975164-3a6d0880-f0ae-11e8-8716-f6027884c651.png)

z12 Scrub c8d7ab
![z12-snowdonia-scrub-only](https://user-images.githubusercontent.com/42757252/48975167-3ccf6280-f0ae-11e8-8a59-c4fc68100204.png)

z12 Heath d1e0b4, scrub c8d7ab
![z12-new-scrub-heath-d1e0b4](https://user-images.githubusercontent.com/42757252/48975171-57094080-f0ae-11e8-92e6-1aa368fa5864.png)

z12 Heath d9deb0, scrub c8d7ab
![z12-new-scrub-heath-d9deb0](https://user-images.githubusercontent.com/42757252/48975169-52448c80-f0ae-11e8-8e49-a14f4310248f.png)

z10 Scrub c8d7ab
![z10-snowdonia-scrub-only](https://user-images.githubusercontent.com/42757252/48975173-61c3d580-f0ae-11e8-8b59-c10bf16223fa.png)

z10 Heath d1e0b4, scrub c8d7ab
![z10-new-scrub-heath-d1e0b4](https://user-images.githubusercontent.com/42757252/48975175-65eff300-f0ae-11e8-9667-f1d2efc615ca.png)

z10 Heath d9deb0, scrub c8d7ab
![z10-new-scrub-heath-d9deb0](https://user-images.githubusercontent.com/42757252/48975185-6ab4a700-f0ae-11e8-80e1-f1f40c205c79.png)

**Comment by turnsole80** [comment]
There's an argument to be made that there should be a heath=* tag. It would like for different heath types and fix the natural=fell issue from previous requests. I have no idea how or where to propose this. Nor do I have the means at the moment to be honest, but please consider it as an option

**Comment by jeisenbe** [comment]
@Chris, if a new tag is needed, please discuss this on the Tagging mailing
list or make a proposal page on the openstreetmap.org wiki.

**Comment by Tomasz-W** [comment]
Propably my last proposition, otherwise I'll end in mantal hospital ^^

`#d0e3b6 `+ 10% vertical dash pattern for heath + `#c8d6ab` for scrub

![scrub1](https://user-images.githubusercontent.com/25656654/48977425-2e774b80-f09a-11e8-9769-e33f1281dfbe.jpg)
![scrub2](https://user-images.githubusercontent.com/25656654/48977426-2e774b80-f09a-11e8-8d2e-6b539b948e07.jpg)
![scrub3](https://user-images.githubusercontent.com/25656654/48977427-2e774b80-f09a-11e8-9843-03dc47fb8317.jpg)
![scrub4](https://user-images.githubusercontent.com/25656654/48977428-2f0fe200-f09a-11e8-8284-dd33772d8eca.jpg)
![scrub5](https://user-images.githubusercontent.com/25656654/48977429-2f0fe200-f09a-11e8-86bf-2d4819d6f8d2.jpg)

**Comment by jeisenbe** [comment]
#d0e3b6 for heath is too similar to grass (#cdebbo), as these tests show.


#d0e3b6 is Lch(88,24,125)

#cdebbo (grass) is Lch(90,32,128)

The lightness is only 2% different and hue is very close; the main
difference is chroma or saturation, but this doesn’t show up well on most
screens.

**Comment by Adamant36** [comment]
To blueish tinted/camo looking. We should just go with #d1e0b4 and scrub 3% darken it. Since both looked good and only Jeisenbe didnt want to go with it.

**Comment by jeisenbe** [comment]
What is your objection to using #d9deb0 for heath, as I’ve tested above?

**Comment by Adamant36** [comment]
Whats your objection to using #d1e0b4 for heath as me and Tomasz-W said? Usually we go with a democratic system to decide things. It means you not get your way sometimes, but that's the trade off of doing thing fairly and listening to other peo

You should do a cut and paste side by side of both of them or like the square colored boxes thing. Its hard to compare them otherwise.

**Comment by Adamant36** [comment]
Also, if we went with your color then it wouldnt be such a stright path to a scrub color that we know would work with streams would it? Or would we still with the heath darkened color?

**Comment by jeisenbe** [comment]
My objection to #d1e0b4 was stated in my previous comment: 

"d1e0b4 is too close to c8d7ab (it's only 3% lighter, and otherwise identical)."

No other two landcover colors are this close. Even leisure, which somewhat overlaps with park, is 5% lighter than the park color, and @Tomasz-W previously said that he thinks this is too similar. 

As the test images above show (See https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-441412852), Heath #d1e0b4 looks too similar to scrub #c8d7ab, especially it low zoom levels, eg z13, z12 and z10. 

z12 Scrub c8d7ab - new scrub color, PR in progress
![z12-snowdonia-scrub-only](https://user-images.githubusercontent.com/42757252/48975167-3ccf6280-f0ae-11e8-8a59-c4fc68100204.png)

z12 Heath d1e0b4, scrub c8d7ab - too similar, and heath is too green
![z12-new-scrub-heath-d1e0b4](https://user-images.githubusercontent.com/42757252/48975171-57094080-f0ae-11e8-92e6-1aa368fa5864.png)

z12 Heath d9deb0, scrub c8d7ab - my new suggestion
![z12-new-scrub-heath-d9deb0](https://user-images.githubusercontent.com/42757252/48975169-52448c80-f0ae-11e8-8e49-a14f4310248f.png)

Another problem, which I forgot to mention, is that bog uses the heath color. Bogs are quite different from marshes; they are made from low-growing mosses which form peat, while marshes have grasses and similar plants. So they use the brownish heath color as a background to distinguish them from marshes, which have the green grass background color. 

If we make heath too green, it will not be easy to distinguish bogs from marshes. 

For both of these reasons, I believe it would be better to give heath a color with a hue more on the yellow side of green, for example #d9deb0 as I tested in the previous comment.

BTW, @Adamant36, could you give me the link to the location you tested in previously in this comment: https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-441060481
It looks like a great place for testing!

**Comment by jeisenbe** [comment]
Here's a better example of the problem with #d1e0b4. At z13 zoom level the landcover colors are the same as at higher zoom levels, but there are no patterns shown.

_[Goodwick, Wales](http://www.openstreetmap.org/#map=13/52.0101/-4.9899), z13 with heath d1e0b4:_
**Can you easily tell scrub from heath with #d1e0b4?** 
![goodwick-z13-heath-d1e0b4](https://user-images.githubusercontent.com/42757252/48988459-01de3500-f169-11e8-8eec-8e39e78ea0eb.png)
 
_How about now?_ Heath #d9deb0
![goodwick-z13-heath-d9deb0](https://user-images.githubusercontent.com/42757252/48988567-87fa7b80-f169-11e8-8ab8-b04b35407b54.png)

**z14 to compare** http://www.openstreetmap.org/#map=14/52.0101/-4.9899
heath #d1e0b4
![goodwick-z14-heath-d1e0b4](https://user-images.githubusercontent.com/42757252/48988933-64d0cb80-f16b-11e8-8383-12a7920b6f12.png)

heath #d9deb0
![z14-goodwick-heath-d9deb0](https://user-images.githubusercontent.com/42757252/48988937-67cbbc00-f16b-11e8-80e7-420aa78c67d7.png)

**Here is an area of bog (on the left side) near marshes (Right, upper and middle)**
http://www.openstreetmap.org/#map=7/52.81443/-4.01972

Current heath rendering (with new scrub):
![bog-marsh-scrub-only](https://user-images.githubusercontent.com/42757252/48988916-5e425400-f16b-11e8-8a01-93b81b2a7824.png)

Can you tell the marshes from the bog with #d1e0b4 for heath and bog? 
![bog-marsh-heath-d1e0b4](https://user-images.githubusercontent.com/42757252/48988665-022b0000-f16a-11e8-88ad-aab8a0af86da.png)

With #d9deb0 for bog and heath, it is easier to see the difference. 
![bog-marsh-heath-d9deb0](https://user-images.githubusercontent.com/42757252/48988912-58e50980-f16b-11e8-800c-1a79115426fd.png)

I also see that with #d9deb0 for heath is **easier to distinguish heath from grass.**

**Comment by jeisenbe** [comment]
_Re: "You should do a cut and paste side by side of both of them or like the square colored boxes thing"_

Here's the two colors, with grass on the left and the new scrub color on the right. 
(The page background is set to land-color; #f2efe9 - light gray)

d1e0b4 (middle)
![grass-heath-d1e0b4-scrub-colorlch](https://user-images.githubusercontent.com/42757252/48989198-8bdbcd00-f16c-11e8-9361-cb227aa23dda.png)

d9deb0 (middle)
![grass-heath-d9deb0-scrub-colorlch](https://user-images.githubusercontent.com/42757252/48989201-8d0cfa00-f16c-11e8-918f-f27d62716e3a.png)

(The color picker is showing the Lch values, but I actually entered the hex codes directly)

**Comment by Adamant36** [comment]
Hhmmm I guess that all makes sense. 3% isnt that much of difference and it probably should be more. The bog thing, I dont know. Is the colors of bogs and heath the same in real life? If not, maybe they should a seperate color. 

If we arent going with d1e0b4 for heath, is there a reason we couldnt use it for the scrub color as it was originally entended? I only said it would work for heath because of it having a second color that we could switch it out with. Now that we arent using it for heath though, id prefer it be the scrub color. It was the better of the two. Although its not that big of a deal, as both colors resolve the stream issue. Which is why we are here. I think....

What are you using to tweak with the colors? Is that an app or a website?

I'll try and find the location from my example. I know it was in the New York area. I want say around Scranton. There isnt much heath tagging around there. So it should be easy enough to find in OverPass Turbo. I think its a good place because of the lake. Whatever new color for heath is picked it should be tested against a large body of water. Since theres a tendecy for colors to washout the blue for some reason.

**Comment by jeisenbe** [comment]
This is the link to the color gradient page. Yse the dropdown menu to change "Colour selection mode" to "show all", and then you can enter hex values or RGB.
You can also set the "Page background colour:" to #f2efe9 (the untagged land color)

http://davidjohnstone.net/pages/lch-lab-colour-gradient-picker#eef0d5,cdebb0,d5d89f,b6e3b5,add19e,aedea3,c8facc,aae0cb,b6e3b5,def6c0,abccb0

**Comment by jeisenbe** [comment]
I believe that this issue has been partially addressed by the changed scrub color. The old scrub color was a bluer shade of green, which contrasted particularly strongly with heath. 

@imagico used slightly different colors for vegetation areas on his branch, recently. I was considering making a PR, but the difference is slight, especially for grass (only 1% lighter) and scrub (a couple percent more chroma and a little shift in hue, on the alt-colors branch). 

Health is the only color that is significantly different on alt-colors; it was changed from `#d6d99f` LCH(85,30,110) to `#dae2ac` LCH(88,28,114)  - still a small change, though the difference in lightness is visible (and probably is why grass was changed to be 1 point lighter):

![lch-alt-vs-current](https://user-images.githubusercontent.com/42757252/51534406-0f721b80-1e89-11e9-9f9d-343f5e7cc6a2.png)

These changes would make the different types of vegetation slightly closer together in color, compared to the current colors; heath is shifted to be a little closer to scrub, which is more similar to forest/wood. But perhaps the reduced contrast is still enough?

A number of renderings are shown on this page:
http://blog.imagico.de/more-on-vegetation-rendering-in-openstreetmap-maps/

**Comment by kocio-pl** [comment]
Looks reasonable to me. I would just tune the wood color to be lighter or just a bit more yellow, because woods tend to make "heavy" areas on the map (large and dark at the same time).

**Comment by kocio-pl** [comment]
I have discussed forest color in https://github.com/gravitystorm/openstreetmap-carto/issues/3513#issuecomment-456731561.
